apps/asap-server/src/middleware/error-handler.ts

```diff
@@ -1,15 +1,18 @@
 import { ErrorRequestHandler } from 'express';
-import debug from 'debug';
 import { isBoom } from '@hapi/boom';
+import { Logger } from 'pino';
 
-const logger = debug('asap-server');
-
-export const errorHandler: ErrorRequestHandler = (err, req, res, next) => {
+export const errorHandlerFactory = (logger: Logger): ErrorRequestHandler => (
```

Comment:
Instead of changing this to a factory. ErrorHandler could easily just use the logger on injected on the request e.q: `req.log`

Reply:
yeah I mean it could.. people often use this DI pattern and attach libs to the request object. But to be fair i prefer this approach as it is much easier to test so I'd prefer to leave it as is